### PR TITLE
Remove GetFlags mixin and deprecated methods from flag model

### PR DESCRIPTION
**Pull request by krauselukas**
The GetFlags mixin was only used by the bento ui and
can be removed.

Related to https://github.com/openSUSE/open-build-service/issues/8314

### CODE REVIEW

**Comment by dmarcoux** [comment]
There are also a few methods the `Flag` model which can be removed since they are only used in Bento and in this `GetFlags` mixin. They are marked with the comment `# TODO: used by bento. Remove when dropping old UI.`

**Comment by krauselukas** [comment]
> There are also a few methods the `Flag` model which can be removed since they are only used in Bento and in this `GetFlags` mixin. They are marked with the comment `# TODO: used by bento. Remove when dropping old UI.`

@dmarcoux yeah I saw them already, will take care of them as well :)

File: src/api/.rubocop.yml [hunk 1]
```diff
@@ -36,6 +36,13 @@ Style/MultilineWhenThen:
 Style/UnneededInterpolation:
   Enabled: false
 
+# Disable this cop for the minitest, because it will be moved to rspec anyway and
+# a couple of outdated bento ui methods need to be moved to the minitest's
+# in order to get removed from the main's project codebase
+Metrics/ClassLength:
+  Exclude:
+    - 'test/unit/project_test.rb'
```
Comment (coolo) [inline src/api/.rubocop.yml]: rather disable in the file - as done in 4 other tests

File: src/api/test/unit/project_test.rb [hunk 2]
```diff
@@ -18,18 +18,47 @@ def test_maintained_project_names
...
+
+    the_flags
+  end
+
   # TODO: bento_only
   def _project_flags(project, flag)
     architectures = Architecture.where(id: project.repository_architectures.select(:architecture_id)).reorder('name').distinct
     repository_names = project.repositories.pluck(:name)
-    project.get_flags(flag, repository_names, architectures)
+    get_flags(project, flag, repository_names, architectures)
```
Comment (coolo) [inline src/api/test/unit/project_test.rb]: get rid of the complete test instead. There is no point in testing a function that only exists in the test itself.

File: src/api/test/unit/project_test.rb [hunk 3]
```diff
@@ -18,18 +18,47 @@ def test_maintained_project_names
...
   end
 
   # TODO: bento_only
   def _package_flags(package, flag)
     architectures = Architecture.where(id: package.project.repository_architectures.select(:architecture_id)).reorder('name').distinct
     repository_names = package.project.repositories.pluck(:name)
-    package.get_flags(flag, repository_names, architectures)
+    get_flags(package, flag, repository_names, architectures)
```
Comment (coolo) [inline src/api/test/unit/project_test.rb]: get rid of the complete test instead

File: src/api/test/unit/project_test.rb [hunk 4]
```diff
@@ -122,7 +151,7 @@ def test_flags_inheritance
     package2.store
 
     # the all 'disable' build flag shall overwrite the project setting
-    package2.get_flags('build', ['SLE_11_SP4'], Architecture.all).each do |repo|
+    get_flags(package2, 'build', ['SLE_11_SP4'], Architecture.all).each do |repo|
```
Comment (coolo) [inline src/api/test/unit/project_test.rb]: same here. Drop all testing of the local function get_flags.

We have a rspec for Flag::SpecifiedFlags and that's what is used.

File: src/api/test/unit/project_test.rb (hunk 2 above)
Comment (krauselukas) [reply src/api/test/unit/project_test.rb]: @coolo I thought the specified_flags rspec test is not covering all the scenarios, but yeah I looked again, and it seems its covering the inheritance of the flags (even though a bit simpler). So I will remove all this stuff.

File: src/api/app/models/flag.rb [hunk 5]
```diff
@@ -38,16 +38,12 @@ def arch
...
-    repo_flag = main_object.flags.find_by('flag = ? AND repo = ? AND architecture_id IS NULL', flag, repo)
-    arch_flag = main_object.flags.find_by('flag = ? AND repo IS NULL AND architecture_id = ?', flag, architecture_id)
-    same_flag = main_object.flags.find_by('flag = ? AND repo = ? AND architecture_id = ?', flag, repo, architecture_id)
-    if main_object.is_a?(Package)
-      same_flag ||= main_object.project.flags.find_by('flag = ? AND repo = ? AND architecture_id = ?', flag, repo, architecture_id)
-      repo_flag ||= main_object.project.flags.find_by('flag = ? AND repo = ? AND architecture_id IS NULL', flag, repo)
-      arch_flag ||= main_object.project.flags.find_by('flag = ? AND repo IS NULL AND architecture_id = ?', flag, architecture_id)
-      all_flag ||= main_object.project.flags.find_by('flag = ? AND repo IS NULL AND architecture_id IS NULL', flag)
-    end
+    pkg_or_prj_flags = main_object.is_a?(Package) ? main_object.project.flags : main_object.flags
```
Comment (coolo) [inline src/api/app/models/flag.rb]: careful. this is not the equivalent code. repo_flag for packag is package.flag || project.flag - you set it to package.flag

Comment (coolo) [reply src/api/app/models/flag.rb]: but the good news: this function is not used anymore - if you cleaned up the unit tests

Comment (krauselukas) [reply src/api/app/models/flag.rb]: Was still used by Specified::Flags rspec test, but I adapted the test, so the default_status method is gone now :)

File: src/api/spec/models/flag/specified_flags_spec.rb [hunk 6]
```diff
@@ -16,8 +16,8 @@
     end
 
     it 'has default status on effective flag' do
-      expect(subject.effective_flag(nil, nil).status).to eq(Flag.default_status(flag_type))
-      expect(subject.effective_flag(repository_name, architecture_name).status).to eq(Flag.default_status(flag_type))
+      expect(subject.effective_flag(nil, nil).status).to eq(subject.default_flag(nil, nil).status)
```
Comment (coolo) [inline src/api/spec/models/flag/specified_flags_spec.rb]: Sorry Lukas, I don't get it. `Flag.default_status(flag_type)` is just a one liner calling `FlagHelper.default_for(flag_name)` . The function you replace it with is a much more complex variant to call the same helper

Comment (krauselukas) [reply src/api/spec/models/flag/specified_flags_spec.rb]: Puhh right, this Flag stuff confuses me, I changed it to use the helper directly, thanks for the hint

Comment (coolo) [reply src/api/spec/models/flag/specified_flags_spec.rb]: https://www.youtube.com/watch?v=LeyofQK6tRw

**Pull request merged**